src/Microsoft.Extensions.ProjectModel.Sources/DotNet/DotNetProjectContextFactory.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.DotNet.ProjectModel;
+
+namespace Microsoft.Extensions.ProjectModel
+{
+    internal class DotNetProjectContextFactory
+    {
+        public IProjectContext Create(string filePath, string configuration)
+        {
+            var project = new ProjectContextBuilder()
+                .AsDesignTime()
```

Comment:
no. this ensures when fragment file is missing the project model will tolerant it. it doesn't add additional references, but allow them to be missing. see here: https://github.com/dotnet/cli/blob/1278e9d6bf35500e9672fe3f53125df1d08717d8/src/Microsoft.DotNet.ProjectModel/Resolution/MSBuildDependencyProvider.cs#L31